tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -0,0 +1,255 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Run kubevirt testing
+# Maintainer: Nan Zhang <nan.zhang@suse.com>
+
+use base multi_machine_job_base;
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use utils;
+use Config::Simple;
+
+my @kubevirt_tests = (
+    'vmi_lifecycle_test',
+    'vm_test',
+    'vmipreset_test',
+    'vmi_configuration_test',
+    'vmi_headless_test',
+    'access_test',
+    'storage',
+    'container_disk_test',
+    'datavolume',
+    'hotplug',
+    'vmi_cloudinit_test',
+    'vmi_iothreads_test',
+    'vmi_multiqueue_test',
+    'primary_pod_network',
+    'vmi_multus',
+    'vmi_networking',
+    'networkpolicy',
+    'vmi_hostdev_test',
+    'vmi_gpu_test',
+    'vmi_monitoring_test',
+    'vnc_test',
+    'console_test',
+    'credentials_test',
+    'expose',
+    'replicaset_test',
+    'network_services',
+    'vmi_kernel_boot_test',
+    'infra_test',
+    'operator_test',
+    'migration_test',
+    'subresource_api_test',
+    'virt_control_plane_test'
+);
+
+sub run {
+    my ($self) = @_;
+    my $ip_out = script_output('ip route show | grep -Eo "src\s+([0-9.]*)\s+" | head -1 | cut -d\' \' -f 2', 30);
+    set_var('SERVER_IP', $ip_out);
+    bmwqemu::save_vars();
+
+    # Wait for destination to be ready
+    $self->set_mutex_lock('DST_READY_TO_START');
+
+    my $agent_ip = $self->get_var_from_parent("AGENT_IP");
+
+    rke2_server_setup($self, $agent_ip);
+    deploy_kubevirt_manifests();
+    run_virt_tests();
+    generate_test_report();
+    upload_results();
+}
+
+# Mutex lock/unlock functionality
+sub set_mutex_lock {
+    my ($self, $lock_signal) = @_;
+
+    mutex_lock($lock_signal);
+    mutex_unlock($lock_signal);
+}
+
+sub rke2_server_setup {
+    my ($self, $agent_ip) = @_;
+
+    systemctl('stop apparmor.service');
+    # systemctl('stop SuSEfirewall2.service');
+    assert_script_run('ssh-keygen -b 2048 -t rsa -q -N "" -f ~/.ssh/id_rsa') unless (-e "~/.ssh/id_rsa");
+    exec_and_insert_password("ssh-copy-id -o StrictHostKeyChecking=no -i /root/.ssh/id_rsa.pub root\@$agent_ip");
+
+    # Ensure SUSE certificates are installed on the node
+    ensure_ca_certificates_suse_installed();
+
+    # Install downstream kubevirt packages
+    zypper_call('in kubernetes1.18-client') if (script_run('rpmquery kubernetes1.18-client'));
+
+    # RKE2 deployment on server node
+    my $rke_ver = get_var('INSTALL_RKE2_VERSION');
```

Comment:
Comparing to setting INSTALL_RKE2_VERSION in both agent and server testsuite, suggest to reuse one in a single test, either server or agent testsuite, to avoid version mismatch.

Reply:
Yes, actually the variable is got from test suite parameters or the test parameters in yaml.

Reply:
Yes, actually the variable is got from test suite parameters or the test parameters in yaml.

Reply:
The point was to get the setting from one single place, or let's say one yaml not in two yamls, to avoid unequal manual configuration issues.

Reply:
By default, it will use the latest version for the server and agent service setup. If it's required, we can just set this parameter in server yaml, and then get its value for agent service setup.

Reply:
I have another idea. Instead of updating yaml file, I think it's easier to update test suite settings in OSD UI. So I prefer to keep this setting INSTALL_RKE2_VERSION in both server and agent. A few code changes will happen.

Reply:
It is similar, no matter it is in yaml or testsuite -- set it in one testsuite/yaml, then use get_var, get_var_from_parent to query the same var source.

Reply:
As Julie's comments (https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/14364#discussion_r816494219), merged into one yaml file to schedule both server and agent tests will resolve this issue.